frontend/src/components/object/DownloadObjectButton.vue

```diff
@@ -0,0 +1,53 @@
+<script setup lang="ts">
+// Types
+import { ButtonMode } from '@/interfaces/common/enums';
+import { ref, type PropType } from 'vue';
+
+// PrimeVue/Font
+import Button from 'primevue/button';
+import Dialog from 'primevue/dialog';
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+// State
+import { useObjectStore } from '@/store/objectStore';
+
+const objectStore = useObjectStore();
+
+// Props
+const props = defineProps({
+  mode: {
+    type: String as PropType<ButtonMode>,
+    required: true,
+  },
+  ids: {
+    type: Array<string>,
+    required: true,
+  },
+});
+
+// Download the object(s)
+const displayNoFileDlg = ref(false);
+const download = () => {
+  if (props.ids.length) {
+    // For now we are looping the supplied IDs (if multiple selected) until there is a batching feature
+    props.ids.forEach((i) => {
+      objectStore.getObject(i, undefined, true);
```

Comment:
i'm guessing this shouldnt be awaiting calls to an async getObject
not sure if or where you want to clear `mutiSelectedObjects ` in the store to uncheck checkboxes after downlads

Reply:
It is true that this is an async function. However, the outcome of this function is fundamentally the invocation of a window.open call. As we don't really have any subsequent logic that depends on the resolution of window.open call, this looks sufficient.

Reply:
No it's just kicking them off and letting them download in their own time. I'll see about clearing those in my next PR when I do the delete too. That's a good point I think.